### PR TITLE
Allow embedded links in preloaded api request

### PR DESCRIPTION
**Pull request by spacedmonkey**
Adding a simple fix for [51722](https://core.trac.wordpress.org/ticket/51722). The `response_to_data` method to resolve the links and embed fields. 

Trac ticket: https://core.trac.wordpress.org/ticket/51722

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by spacedmonkey** [comment]
@TimothyBJacobs I added unit tests now.

File: src/wp-includes/rest-api.php [hunk 1]
```diff
@@ -2512,19 +2512,18 @@ function rest_preload_api_request( $memo, $path ) {
 	}
 
 	$request = new WP_REST_Request( $method, $path_parts['path'] );
+	$embed   = false;
 	if ( ! empty( $path_parts['query'] ) ) {
+		$query_params = array();
 		parse_str( $path_parts['query'], $query_params );
+		$embed = isset( $query_params['_embed'] ) ? $query_params['_embed'] : false;
```
Comment (spacedmonkey) [inline src/wp-includes/rest-api.php:2519]:
```suggestion
		$embed = isset( $query_params['_embed'] ) ? rest_parse_embed_param( $query_params['_embed'] ) : false;
```

Comment (spacedmonkey) [reply src/wp-includes/rest-api.php:2519]: We could do this.

**Comment by TimothyBJacobs** [comment]
Thanks for working on this, I combined this with @lukaspawlik's previous PR in https://github.com/WordPress/wordpress-develop/commit/bb395706f4ebe6efeca2a951fad69ba4d98fb799.

**Pull request closed without merging**